app/src/js/reducers/granules.js

```diff
@@ -64,159 +59,81 @@ export const initialState = {
-    set(state, ['bulk', config.requestId, 'error'], action.error);
-  },
-
-  [GRANULE_DELETE]: (state, action) => {
-    const { id } = action;
-    set(state, ['deleted', id, 'status'], 'success');
-    set(state, ['deleted', id, 'error'], null);
-  },
-  [GRANULE_DELETE_INFLIGHT]: (state, action) => {
-    const { id } = action;
-    set(state, ['deleted', id, 'status'], 'inflight');
-  },
-  [GRANULE_DELETE_ERROR]: (state, action) => {
-    const { id } = action;
-    set(state, ['deleted', id, 'status'], 'error');
-    set(state, ['deleted', id, 'error'], action.error);
-  },
+    state.list.inflight = false;
+    state.list.error = action.error;
+  },
+
+  [GRANULE_REPROCESS]: createSuccessReducer('reprocessed'),
+  [GRANULE_REPROCESS_INFLIGHT]: createInflightReducer('reprocessed'),
+  [GRANULE_REPROCESS_ERROR]: createErrorReducer('reprocessed'),
+  [GRANULE_REINGEST]: createSuccessReducer('reingested'),
+  [GRANULE_REINGEST_INFLIGHT]: createInflightReducer('reingested'),
+  [GRANULE_REINGEST_ERROR]: createErrorReducer('reingested'),
+  [GRANULE_APPLYWORKFLOW]: createSuccessReducer('executed'),
+  [GRANULE_APPLYWORKFLOW_INFLIGHT]: createInflightReducer('executed'),
+  [GRANULE_APPLYWORKFLOW_ERROR]: createErrorReducer('executed'),
+  [GRANULE_REMOVE]: createSuccessReducer('removed'),
+  [GRANULE_REMOVE_INFLIGHT]: createInflightReducer('removed'),
+  [GRANULE_REMOVE_ERROR]: createErrorReducer('removed'),
+  [BULK_GRANULE]: createSuccessReducer('bulk', getConfigRequestId),
+  [BULK_GRANULE_INFLIGHT]: createInflightReducer('bulk', getConfigRequestId),
+  [BULK_GRANULE_ERROR]: createErrorReducer('bulk', getConfigRequestId),
+  [GRANULE_DELETE]: createSuccessReducer('deleted'),
+  [GRANULE_DELETE_INFLIGHT]: createInflightReducer('deleted'),
+  [GRANULE_DELETE_ERROR]: createErrorReducer('deleted'),
 
   [SEARCH_GRANULES]: (state, action) => {
-    set(state, ['list', 'params', 'prefix'], action.prefix);
+    state.list.params.prefix = action.prefix;
   },
-  [CLEAR_GRANULES_SEARCH]: (state, action) => {
-    set(state, ['list', 'params', 'prefix'], null);
+  [CLEAR_GRANULES_SEARCH]: (state) => {
+    delete state.list.params.prefix;
   },
-
   [FILTER_GRANULES]: (state, action) => {
-    set(state, ['list', 'params', action.param.key], action.param.value);
+    state.list.params[action.param.key] = action.param.value;
   },
   [CLEAR_GRANULES_FILTER]: (state, action) => {
-    set(state, ['list', 'params', action.paramKey], null);
+    delete state.list.params[action.paramKey];
   },
-
   [OPTIONS_COLLECTIONNAME]: (state, action) => {
     const options = action.data.results.reduce(
       (obj, { name, version }) =>
         Object.assign(obj, {
-          [`${name} ${version}`]: getCollectionId({ name, version })
+          [`${name} ${version}`]: getCollectionId({ name, version }),
         }),
       {}
     );
-    set(state, ['dropdowns', 'collectionName', 'options'], options);
+    state.dropdowns.collectionName = { options };
```

Comment:
In this case, the purpose of the options is strictly for holding the dropdown options for a specific collection, so there are no other properties associated it. However, I'll restore the previous behavior, just in case other properties might be added later.

Reply:
Ah ok, I could have accepted either answer, thanks for knowing it.